docs/rfc/identity_proofs.md

```diff
@@ -0,0 +1,214 @@
+
+The actual proof consists of the following tuple:
+
+    (root, revision, public key, signature)
+
+> TODO: include a protocol bit (denoting "git"), for forwards compat?
+
+The values `root`, `revision`, and `public key` are specified in
+[identities][ids], and it is RECOMMENDED to follow the serialisation formats
+devised there. `signature` is the Ed25519 signature over `revision`, in much the
+same way as the actual revision is signed. All values can thus be obtained by
+inspecting the identity storage.
+
+It is beyond the scope of this document to devise the exact external format to
+serialise the tuple into, as this is expected to vary from service to service.
+
+## Revocation
+
+A claim can be revoked by creating a new identity revision which simply does not
+contain the claim payload. Likewise, a later claim describing the same `(service
+name, account)` mapping invalidates an earlier one.
+
+> NOTE: claiming multiple accounts per service is permissible in principle, but
+> might be undesirable for certain services. In this case, service-specific
+> post-validation is required.
+
+## Verification
+
+1. Given the 4-tuple specified [above][#proof-generation], it is first verified
+   that the signature is valid for the given `revision` and `public key`.
+
+2. If it is, the identity history needs to be resolved from local storage, or
+   the network.
+
+  2.1 If persistence of verification results is available, the identity document
+      at the given `revision` is retrieved, its `root` compared against the
+      supplied one, and it is verified that the given `public key` is a member
+      of the `delegations` set. If expiration is specified in the claim, it is
+      determined if `expiration` seconds already elapsed since `created`
+      according to the clock, in which case verification fails.
+  2.2 Otherwise, the identity verification procedure is run first, after which
+      the history is walked backwards until the `revision` is found. Then, step
+      2.1 is performed.
+
+3. Lastly, the identity history needs to be examined for revocations.
+
+## Discussion
+
+The inclusion of the `revision` in the proof allows to assert that `root` is
+indeed an ancestor, which opens up another way to detect "forks" of the identity
+history: due to the peer-to-peer nature of the `radicle-link` network, it is
+vulnerable to attacks which involve withholding data from other participants, in
+which case a fork may go unnoticed.
+
+It should be noted, however, that refreshing the proof from time to time in
+order to ensure freshness of the data retrieved through `radicle-link` is not
+always practicable.
+
+In order to prove that the(ir own) server is not lying by omission, Keybase
+[anchors a merkle root][keybase-stellar] on a blockchain, which includes all
+sigchains registered in the Keybase directory. Because `radicle-link` does not
+have such a central directory, this approach could only be applied to a partial
+view of the network.
```

Comment:
Something for us to consider?

Reply:
Ceramic does something like this, where people can batch their transactions into a larger one (and share tx fees? Idk). Not sure where you get the merkle proof from.